### ISSUE TITLE
Detect when user went offline 

### ISSUE THREAD

**Issue by TitanNano** [zhukov/webogram#1198]
This simply means,
make use of `navigator.onLine` and the `online` and `offline` events which are emitted on `document.body` and bubble up to `window`. 

It's quite battery hungry to constantly make new HTTP requests when the user is clearly offline, so this would be a good improvement. 
Also we could just show an offline status instead of "connecting..." and "waiting for connection..."

**Comment by Ryuno-Ki** [comment]
As far as I know, [the implementation of navigator.onLine differs between browsers](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine).

**Comment by Ryuno-Ki** [comment]
My comment was meant as "Don't rely only on the event resp. `onLine", but have fallbacks in place.

**Comment by TitanNano** [comment]
yes but we this ticket was meant as an additional feature. None of the current behavior should be removed.

**Comment by zhukov** [comment]
We actually use `online` event handler on `<body>`:
https://github.com/zhukov/webogram/blob/master/app/js/lib/mtproto.js#L941

But we use it to just force check connection callback. Otherwise we retry network request every this number of seconds:
```
this.checkConnectionPeriod = Math.min(30, (1 + this.checkConnectionPeriod) * 1.5)
```

**Comment by TitanNano** [comment]
@zhukov I see. But constantly checking for a connection when the user is offline starts to drain battery. I noticed that both on my laptop and my phone.
